src/components/implementation/no_interface/cFE_booter/osfilesys.h

```diff
@@ -80,34 +112,126 @@ static inline uint32 oct_to_dec(char *oct)
     return tot;
 }
 
+//this could be more performant bithack
+static inline uint32 round_to_blocksize(uint32 offset)
+{
+    if (offset % TAR_BLOCKSIZE) return offset + (TAR_BLOCKSIZE - (offset % TAR_BLOCKSIZE));
+    return offset;
+}
+
+
+static inline char *path_to_name(char *path){
+
+    uint32 path_len = strlen(path), offset;
+    assert(path[path_len - 2] != '/' && path_len > 1);
+
+    // offset is second to last char in path, because last char may be trailing /
+    for ( offset = path_len - 2 ; path[offset] != '/' && offset > 0 ; offset--) {
+        //do nothing
+    }
+    assert(0 < strlen(path + offset + 1));
+    return path + offset + 1;
+}
+
 /******************************************************************************
 ** fsobj Level Methods
 ******************************************************************************/
+static inline int32 rm(struct fsobj *o)
+{
+    assert(o && o->child == NULL);
+
+    // if o is first in list of children
+    if (o->prev == NULL && o->parent) {
+        assert(o->parent->child == o);
+        // if next = null this still works
+        o->parent->child = o->next;
+    }
 
-uint32 fsobj_init(struct fsobj *o);
+    // these still work if next or prev = null
+    if (o->prev) o->prev->next = o->next;
+    if (o->next) o->next->prev = o->prev;
+
+    //we do not do deallocate file data but we do reuse fsobj
+    o->name = NULL;
+    o->filedes = 0;
+    o->size = 0;
+    o->refcnt = 0;
+    o->mode = 0;
+    o->file_part = NULL;
+    return OS_FS_SUCCESS;
+}
```

Comment:
Why whats the drawback to using too many static inline?  Object file size?  But I guess it would make more sense to put these methods inside of osfilesys.c rather than declare in osfilesys.h